protocol/messenger_contact_verification.go

```diff
@@ -0,0 +1,726 @@
+	}
+
+	if persistedVR == nil {
+		// This is a response for which we have not received its request before
+		persistedVR = &verification.Request{}
+		persistedVR.From = contactID
+		persistedVR.To = myPubKey
+	}
+
+	persistedVR.RequestStatus = verification.RequestStatusDECLINED
+	persistedVR.RepliedAt = request.Clock
+
+	err = m.verificationDatabase.SaveVerificationRequest(persistedVR)
+	if err != nil {
+		m.logger.Debug("Error storing verification request", zap.Error(err))
+		return err
+	}
+
+	err = m.SyncVerificationRequest(context.Background(), persistedVR)
+	if err != nil {
+		return err
+	}
+
+	state.AllVerificationRequests = append(state.AllVerificationRequests, persistedVR)
+
+	// TODO: create or update activity center notification
+
+	return nil
+}
+
+func (m *Messenger) HandleContactVerificationTrusted(state *ReceivedMessageState, request protobuf.ContactVerificationTrusted) error {
+	if common.IsPubKeyEqual(state.CurrentMessageState.PublicKey, &m.identity.PublicKey) {
+		return nil // Is ours, do nothing
+	}
+
+	myPubKey := hexutil.Encode(crypto.FromECDSAPub(&m.identity.PublicKey))
+	contactID := hexutil.Encode(crypto.FromECDSAPub(state.CurrentMessageState.PublicKey))
+
+	contact := state.CurrentMessageState.Contact
+	if !contact.Added || !contact.HasAddedUs {
+		m.logger.Debug("Received a verification trusted for a non mutual contact", zap.String("contactID", contactID))
+		return errors.New("must be a mutual contact")
+	}
+
+	err := m.verificationDatabase.SetTrustStatus(contactID, verification.TrustStatusTRUSTED, m.getTimesource().GetCurrentTime())
+	if err != nil {
+		return err
+	}
+
+	err = m.SyncTrustedUser(context.Background(), contactID, verification.TrustStatusTRUSTED)
+	if err != nil {
+		return err
+	}
+
+	persistedVR, err := m.verificationDatabase.GetVerificationRequestFrom(contactID)
+	if err != nil {
+		m.logger.Debug("Error obtaining verification request", zap.Error(err))
+		return err
+	}
+
+	if persistedVR != nil && persistedVR.RepliedAt > request.Clock {
+		return nil // older message, ignore it
+	}
+
+	if persistedVR.RequestStatus == verification.RequestStatusCANCELED {
+		return nil // Do nothing, We have already cancelled the verification request
+	}
+
+	if persistedVR == nil {
+		// This is a response for which we have not received its request before
+		persistedVR = &verification.Request{}
+		persistedVR.From = contactID
+		persistedVR.To = myPubKey
+	}
+
+	persistedVR.RequestStatus = verification.RequestStatusTRUSTED
+
+	err = m.verificationDatabase.SaveVerificationRequest(persistedVR)
+	if err != nil {
+		m.logger.Debug("Error storing verification request", zap.Error(err))
+		return err
+	}
+
+	err = m.SyncVerificationRequest(context.Background(), persistedVR)
+	if err != nil {
+		return err
+	}
+
+	state.AllVerificationRequests = append(state.AllVerificationRequests, persistedVR)
+
+	contact.VerificationStatus = VerificationStatusVERIFIED
```

Comment:
I have a small issue where at the end of the process, the request is marked as trusted, but the verificationStatus stays at 0 when I look at it on the client side, maybe you'll be able to help me @richard-ramos

Reply:
Is this still relevant?

Reply:
Ah my bad, Richard helped me fix that